src/runtime_bignum.nr

```diff
@@ -298,35 +305,35 @@ impl<let N: u32, Params> BigNumInstanceTrait<BigNum<N, Params>> for BigNumInstan
     // ####################################################################################################################
 
     fn __neg(self, val: BigNum<N, Params>) -> BigNum<N, Params> {
-        self.__neg_impl(val)
+        unsafe{ self.__neg_impl(val) }
```

Comment:
This shouldn't be done and the trait method should be marked unconstrained instead.